src/email/services/mailing.service.ts

```diff
@@ -19,4 +26,31 @@ export class MailingService {
       throw new MailSendingException();
     }
   }
+
+  public async sendMailsToSubscribers(): Promise<void> {
+    const { currencyCode, date, rate } =
+      await this.rateService.getExchangeRate();
```

Comment:
I think it's better to move this responsibility to another service. The simplest way is to move getExchangeRate(); into public async scheduleEveryDayTask(): Promise<void>, and pass this data to the method through arguments

Reply:
Good catch, thanks!